src/components/ToDoList.vue

```diff
@@ -1,35 +1,46 @@
 <template>
-  <li v-for="(todo, index) in toDos" v-bind:key="todo.id">
-    <div class="p-6 bg-gray-200 rounded-lg border-radius">
-      <div class="color #000000">{{ todo.title }}</div>
-      <div class="color #000000">{{ todo.text }}</div>
-      <div class="color #000000">{{ todo.priority }}</div>
-      <div class="color #000000">{{ todo.created_at }}</div>
-      <div class="color #000000">{{ todo.status }}</div>
-    </div>
-
-    <button class="border-radius: 50%;" @click="removeToDo(index)">
-      delete
-    </button>
-  </li>
-  <div></div>
-
-  <img />
+  <div v-for="(todo, index) in toDos" v-bind:key="todo.id">
+    <ToDoItem
+      v-if="editingIndex !== index"
+      @toDoDeleted="removeToDo(index)"
+      :toDo="todo"
+      @click="editingIndex = index"
+    />
+    <ToDoItemEdit
+      v-else
+      @toDoDeleted="removeToDo(index)"
+      @toDoSaved="saveToDo(index)"
+      :toDo="todo"
+      @toDoUpdated="
+        (newTodo) => {

```

Comment:
create a separate function for this action, you might have to add more functionality to it later, and also that would comply with the code guidline, where it says to not write expressions in template except for simple conditions